### PR TITLE
Dk/pruning v5

### PR DESCRIPTION
**Pull request by danking**

### CODE REVIEW

**Comment by danking** [comment]
rob's comment from the last PR. (1) should be somewhat addressed here. (2) is not.

> I think we need to do more here
> 
> 1. Prune splits based on stats results. There's no point building all the masks for reading if we know they're going to be thrown away.
> 2. We need to handle A op B, i.e. A || B this will not get pushed down right now and we will never prune on it

File: vortex-file/src/read/layouts/chunked.rs [hunk 1]
```diff
@@ -77,13 +81,38 @@ impl ChunkedLayout {
         }
     }
 
+    fn metadata_layout(&self) -> VortexResult<Option<Box<dyn LayoutReader>>> {
```
Comment (AdamGS) [inline vortex-file/src/read/layouts/chunked.rs:84]: I need very similar logic to expose statistics to datafusion, WDYT about splitting it out to a different PR we can quickly merge?

Comment (robert3005) [reply vortex-file/src/read/layouts/chunked.rs:84]: How does this function know where in the layout tree it is?

Comment (robert3005) [reply vortex-file/src/read/layouts/chunked.rs:84]: oh, this method is local, eh ignore me

Comment (danking) [reply vortex-file/src/read/layouts/chunked.rs:84]: @AdamGS Yeah I can make PR

Comment (danking) [reply vortex-file/src/read/layouts/chunked.rs:84]: https://github.com/spiraldb/vortex/pull/1399

**Pull request closed without merging**